### PR TITLE
Chore/setup express typescript

### PR DESCRIPTION
**Pull request by taralamia**

### CODE REVIEW

**Comment by parvez-ahammed** [comment]
Put the dist folder in. gitignore. It should not be pushed to the remote repo. Also try to understand what is a dist repo, its purpose and why it should be kept in `.gitignore`

**Comment by parvez-ahammed** [comment]
Put the dist folder in. gitignore. It should not be pushed to the remote repo. Also try to understand what is a dist repo, its purpose and why it should be kept in `.gitignore`

**Comment by parvez-ahammed** [comment]
Same goes for `node_modules`. Keep it in `.gitignore`.

Try to research a little bit about why it is expected to be ignored from the remote repo. A plain explanation is it is a large directory and it doesn't have to be tracked. Each individual can have their own set of `node_modules`. It will be consistent as long as all the packages are installed from the configuration mentioned in the `package.json` and `package-lock.json`.

So the rule of thumb is don't keep sensitive,environment specific data in the remote repo.

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (parvez-ahammed) [inline .vscode/launch.json]: `.vscode` folder can be kept on the `.gitignore` unless necessary to share with a co worker and everyone is expected to have different settings

File: src/chicken.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export class Chicken {
+  cluck() {
```
Comment (parvez-ahammed) [inline src/chicken.ts]: In  a PR no practice code should be provided and it is strictly discouraged. If it is a must to keep unnecessary code then keep it with a comment about  the reason why it should not be deleted.

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import express, { Express, Request, Response } from "express";
+const port = 8000;
+const app: Express = express();
+app.get("/", (req: Request, res: Response) => {
+  res.send("Hello prints in browser too");
```
Comment (parvez-ahammed) [inline src/index.ts]: In here , routes should be used.

Request should go through 

```

routes -> controller  -> service -> repository


For the time being you can keep any in memory DS instead of DB connection for the repository layer

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,113 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig to read more about this file */
+
```
Comment (parvez-ahammed) [inline tsconfig.json]: Remove commented out parts and only keep the things that are necessary and you properly understand.

File: src/server.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+
```
Comment (parvez-ahammed) [inline src/server.ts]: Don't keep any unnecessary file in a PR

**Comment by parvez-ahammed** [comment]
Check all the comments carefully and resolve them one by one and once done again request for a review from here

@taralamia

**Review by parvez-ahammed — CHANGES_REQUESTED** [review]

**Comment by parvez-ahammed** [comment]
- Currently there is no Eslint and prettier setup. 
- Add at least one testing end using the current pattern and dont keep any file empty.

**Comment by taralamia** [comment]
I have setup prettier-eslint. Initially I configured the database connection with mongoDb to check at least one testing end.

File: eslint.config.mjs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  plugins: ["@typescript-eslint"],
+  extends: [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended"
+  ],
+  rules: {
+    "no-unused-vars": "error",
+    "no-undef": "error",
+    "prefer-const": "error",
+    "no-console": "warn"
```
Comment (parvez-ahammed) [inline eslint.config.mjs]: Why we are using this? What would happen if we don't use this?

File: package.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "quash-backend",
+  "version": "1.0.0",
+  "description": "��#\u0000 \u0000Q\u0000u\u0000a\u0000s\u0000h\u0000 \u0000B\u0000a\u0000c\u0000k\u0000e\u0000n\u0000d\u0000\r\u0000 \u0000",
```
Comment (parvez-ahammed) [inline package.json]: Fix the description

File: package.json [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    "prettier": "prettier --ignore-path .gitignore --write \"./src/**/*.+(js|ts|json)\"",
+    "prettier:fix": "prettier --write src"
+  },
+  "devDependencies": {
+    "@eslint/js": "^9.28.0",
+    "@typescript-eslint/eslint-plugin": "^8.33.1",
+    "@typescript-eslint/parser": "^8.33.1",
+    "eslint": "^9.28.0",
+    "nodemon": "^3.1.10",
+    "prettier": "^3.5.3",
```
Comment (parvez-ahammed) [inline package.json:35]: Create a prettier setup file which will override the default prettier setup of the user. Also make sure to ignore dist and node_modules from prettier else it will increase the time to scan. So update the setup according to that.

File: src/app.ts [hunk 9]
```diff
@@ -0,0 +1,8 @@
+import express, { Request, Response } from "express";
+const app = express();
+app.use(express.json());
+app.get('/', (req: Request, res: Response) => {
+    
+  res.send('Hello from setup file');
```
Comment (parvez-ahammed) [inline src/app.ts]: Unnecessary whitespace above. Keep the coding style consistant.

File: src/config/db.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import dotenv from 'dotenv';
+import mongoose from 'mongoose';
+dotenv.config();
+
+const db_url: string = process.env.DB_URL as string;
+ const port : string = process.env.PORT as string;
```
Comment (parvez-ahammed) [inline src/config/db.ts]: Space of left. Is prettier working?

File: src/server.ts [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import express, { Request, Response } from 'express';
+import {port,connectToDatabase} from './config/db';
+const app = express();
+connectToDatabase();
+app.get('/', (req: Request, res: Response) => {
+    res.send('Hello, TypeScript Express!');
+ });
```
Comment (parvez-ahammed) [inline src/server.ts]: Space on left.

**Review by parvez-ahammed — CHANGES_REQUESTED** [review]

File: eslint.config.mjs (hunk 6 above)
Comment (taralamia) [reply eslint.config.mjs]: If we allow the no-unused-vars, no-undef then in our project we'll have undeclared and unused variables. and to prevent bug and maintaining code readability we prefer const.
